codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolTestGenerator.java

```diff
@@ -0,0 +1,572 @@
+        this.service = context.getService();
+        this.protocolName = context.getProtocolName();
+        this.symbolProvider = context.getSymbolProvider();
+        this.writer = context.getWriter();
+
+        this.requestTestBuilder = requestTestBuilder;
+        this.responseTestBuilder = responseTestBuilder;
+        this.responseErrorTestBuilder = responseErrorTestBuilder;
+    }
+
+    public void generateProtocolTests() {
+        OperationIndex operationIndex = model.getKnowledge(OperationIndex.class);
+        TopDownIndex topDownIndex = model.getKnowledge(TopDownIndex.class);
+
+        for (OperationShape operation : new TreeSet<>(topDownIndex.getContainedOperations(service))) {
+            if (operation.hasTag("server-only")) {
+                continue;
+            }
+
+            // 1. Generate test cases for each request.
+            operation.getTrait(HttpRequestTestsTrait.class).ifPresent(trait -> {
+                // TODO filter test cases for this given protocol.
+                requestTestBuilder.build(model, symbolProvider, protocolName, operation,
+                        trait.getTestCases()).generateTestFunction(writer);
+            });
+
+            // 2. Generate test cases for each response.
+            operation.getTrait(HttpResponseTestsTrait.class).ifPresent(trait -> {
+                // TODO filter test cases for this given protocol.
+                responseTestBuilder.build(model, symbolProvider, protocolName, operation,
+                        trait.getTestCases()).generateTestFunction(writer);
+            });
+
+            // 3. Generate test cases for each error on each operation.
+            for (StructureShape error : operationIndex.getErrors(operation)) {
+                if (error.hasTag("server-only")) {
+                    continue;
+                }
+
+                error.getTrait(HttpResponseTestsTrait.class).ifPresent(trait -> {
+                    // TODO filter test cases for this given protocol.
+                    responseErrorTestBuilder.build(model, symbolProvider, protocolName, operation, error,
+                            trait.getTestCases()).generateTestFunction(writer);
+                });
+            }
+        }
+
+        // Include any additional stubs required.
+        for (String additionalStub : additionalStubs) {
+            writer.write(IoUtils.readUtf8Resource(getClass(), additionalStub));
+        }
+    }
+
+    // Only generate test cases when its protocol matches the target protocol.
+    private <T extends HttpMessageTestCase> void onlyIfProtocolMatches(T testCase, Runnable runnable) {
+        if (testCase.getProtocol().equals(settings.getProtocol())) {
+            LOGGER.fine(() -> format("Generating protocol test case for %s.%s", service.getId(), testCase.getId()));
+            runnable.run();
+        }
+    }
+
+//    public void generateRequestTests(List<HttpRequestTestCase> testCases, OperationShape operation) {
```

Comment:
Looks like a ton of code got left behind?

Reply:
thanks! yeah this was forgotten code that should of been deleted.